src/synpp/pipeline.py

```diff
@@ -481,15 +480,16 @@ def update_json(meta, working_directory):
     shutil.move("%s/pipeline.json.new" % working_directory, "%s/pipeline.json" % working_directory)
 
 
-def run(definitions, config = {}, working_directory = None, flowchart_path = None, dryrun = False, verbose = False, logger = logging.getLogger("synpp")):
+def run(definitions, config = {}, working_directory = None, flowchart_path = None, dryrun = False, verbose = False, logger = logging.getLogger("synpp"), rerun_required=True):
     # 0) Construct pipeline config
     pipeline_config = {}
     if "processes" in config: pipeline_config["processes"] = config["processes"]
     if "progress_interval" in config: pipeline_config["progress_interval"] = config["progress_interval"]
 
     if not working_directory is None:
         if not os.path.isdir(working_directory):
-            raise PipelineError("Working directory does not exist: %s" % working_directory)
+            logger.warning("Working directory does not exist, it will be created: %s" % working_directory)
+            os.mkdir(working_directory)
```

Comment:
I see the point, but so far I was deliberately avoiding this, to make sure that the user is 100% aware where all the caching files will go. Maybe we can add a config flag, which is usually "false" like "auto_generate_working_directory" ?